Buddies.API/Controllers/ProjectsController.cs

```diff
@@ -243,7 +245,209 @@ public async Task<ActionResult<SearchResponse>> GetUser(string search, int page,
             return Ok(response);
         }
 
-             
+        /// <summary>
+        /// API route GET /api/v1/projects/:id for fetching project profile.
+        /// </summary>
+        [HttpGet("{id}")]
+        public async Task<ActionResult> GetProfile(int id)
+        {
+            var project = await _context.Projects.FindAsync(id);
+            if (project == null)
+            {
+                return NotFound("PROJECT PROFILE NOT FOUND");
+            }
+            var owner = await _context.Profiles.FindAsync(project.OwnerId);
+            if (owner == null)
+            {
+                return NotFound("OWNER OF PROJECT PROFILE NOT FOUND");
+            }
+            var ownerUser = await _context.Users.FindAsync(project.OwnerId);
+            if (ownerUser == null)
+            {
+                return NotFound("OWNER OF PROJECT PROFILE NOT FOUND");
+            }
```

Comment:
I'm not sure if this is really the best implementation. If you define the owner as a Profile/User object in Project (following [EF docs](https://docs.microsoft.com/en-us/ef/core/modeling/relationships?tabs=fluent-api%2Cfluent-api-simple-key%2Csimple-key) + [loading it properly](https://docs.microsoft.com/en-us/ef/core/querying/related-data/)) then require an owner to be passed in Project's constructor we can avoid these cases.

Reply:
this is a great idea!